pinot-controller/src/main/resources/app/utils/Utils.tsx

```diff
@@ -345,6 +346,28 @@ const splitStringByLastUnderscore = (str: string) => {
   return [beforeUnderscore, afterUnderscore];
 }
 
+export const getDisplaySegmentStatus = (idealState, externalView): DISPLAY_SEGMENT_STATUS => {
+  const externalViewStatesArray = Object.values(externalView || {});
+
+  // if EV is empty or EV contains ERROR state then segment is in Bad state
+  if(externalViewStatesArray.length === 0 || externalViewStatesArray.includes(SEGMENT_STATUS.ERROR)) {
+    return DISPLAY_SEGMENT_STATUS.BAD;
```

Comment:
why don't we make this even clearer?
```
if(externalViewStatesArray.length === 0) {
  return DISPLAY_SEGMENT_STATUS.MISSING
}

if (externalViewStatesArray.includes(SEGMENT_STATUS.ERROR)) {
  return DISPLAY_SEGMENT_STATUS.ERROR
}
```
no need to force people to mouse over to see why it's actually bad.

Reply:
makes sense. added separate MISSING status.
The statuses are self-explanatory, those tooltips are just for convenience.

Reply:
I was thinking ERROR would be better here, but since there's no other "BAD", then this is fine.